src/transforms/wasm.rs

```diff
@@ -0,0 +1,197 @@
+use super::Transform;
+use crate::{
+    event::Event,
+    topology::config::{DataType, TransformConfig, TransformContext, TransformDescription},
+    wasm::{WasmModule, WasmModuleConfig},
+};
+use serde::{Deserialize, Serialize};
+use std::path::PathBuf;
+use vector_wasm::Role;
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct WasmConfig {
+    pub module: PathBuf,
+}
+
+impl Into<WasmModuleConfig> for WasmConfig {
+    fn into(self) -> WasmModuleConfig {
+        WasmModuleConfig::new(Role::Transform, self.module)
+    }
+}
+
+inventory::submit! {
+    TransformDescription::new_without_default::<WasmConfig>("wasm")
+}
+
+#[typetag::serde(name = "wasm")]
+impl TransformConfig for WasmConfig {
+    fn build(&self, _cx: TransformContext) -> crate::Result<Box<dyn Transform>> {
+        Ok(Box::new(Wasm::new(self.clone())?))
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn output_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn transform_type(&self) -> &'static str {
+        "wasm"
+    }
+}
+
+#[derive(Debug)]
+pub struct Wasm {
+    module: WasmModule,
```

Comment:
Since we'll want WASM modules to be able to define their own options, it seems like it's reasonable if we provide an `options: HashMap<String, Value>` that the wasm module gets as a context.